test_util/src/lib.rs

```diff
@@ -808,6 +808,11 @@ async fn main_server(
       *res.status_mut() = StatusCode::FOUND;
       Ok(res)
     }
+    (_, "/server_error") => {
+      let mut res = Response::new(Body::empty());
+      *res.status_mut() = StatusCode::INTERNAL_SERVER_ERROR;
+      Ok(res)
+    }
```

Comment:
Should we add an endoint that errors the first time and succeeds the second?

Reply:
It's gonna be hard to add state to that server to verify how many times this function was called...